flask_simplelogin/__init__.py

```diff
@@ -126,26 +129,40 @@ def wrap(*args, **kwargs):
     return decorator
 
 
+class Message():
+    def __init__(self, text, category="primary", enabled=True):
+        self.text = text
+        self.category = category
+        self.enabled = True
+
+    def flash(self):
+        if self.text and self.enabled:
+            flash(self.text, self.category)
+
+
 class SimpleLogin(object):
     """Simple Flask Login"""
 
     messages = {
-        'login_success': 'login success!',
-        'login_failure': 'invalid credentials',
-        'is_logged_in': 'already logged in',
-        'logout': 'Logged out!',
-        'login_required': 'You need to login first',
-        'access_denied': 'Access Denied',
-        'auth_error': 'Authentication Error: {0}'
+        'login_success': Message('login success!', category='success'),
+        'login_failure': Message('invalid credentials', category='danger'),
+        'is_logged_in': Message('already logged in', category='primary'),
+        'logout': Message('Logged out!', category='primary'),
+        'login_required': Message('You need to login first', category='warning'),
+        'access_denied': Message('Access Denied'),
+        'auth_error': Message('Authentication Error: {0}')
     }
 
     @staticmethod
     def get_message(message, *args, **kwargs):
         """Helper to get internal messages outside this instance"""
-        msg = current_app.extensions['simplelogin'].messages.get(message)
-        if msg and (args or kwargs):
-            return msg.format(*args, **kwargs)
-        return msg
+        msg = current_app.extensions['simplelogin'].messages.get(
+            message)
+        if msg.enabled:
+            if msg.text and (args or kwargs):
+                return msg.text.format(*args, **kwargs)
+            return msg.text
+        return
```

Comment:
What about using [_the flat success path_ approach](https://www.vinta.com.br/blog/2018/flat-success-path/) here to avoid _cyclomatic complexity_ (that is to say, to avoid many `if` and indentation?

I'd go for something like:

```python
        if not msg.enabled or not.text:
            return

        if args or kwargs:
            return msg.text.format(*args, **kwargs)

        return msg.text
```

Reply:
Sounds good!